crates/shared/src/http_solver_api/model.rs

```diff
@@ -130,6 +130,14 @@ pub struct FeeModel {
     pub token: H160,
 }
 
+#[serde_as]
+#[derive(Clone, Debug, Deserialize)]
+pub struct InteractionData {
+    pub target: H160,
+    pub value: U256,
+    pub call_data: Bytes<Vec<u8>>,
+}
```

Comment:
While it's of course possible to encode the entire settlement as one big `interaction_data`, I wonder if it might be nice to still allow mix and match of AMM execution and additional custom calls. For this the interactions would also have to contain an execution plan coordinate and we should also append them to the encoder in their right position (with regards to other AMM interactions).

Wdyt?

Reply:
I implemented your suggestion in the next commit. Please let me know whether the interface looks okay, then I will add tests and clean it up.

I tried to implement your suggestion in a non-breaking way: The `SettledBatchAuctionModel` just go the new field interactions_data. The downside is that the "coordinate" is hard to understand, as now, we have a coordinate in amms and one in interaction_data.

 We could also do, in order to make the "coordinate" of the execution clearer:
```
enum Execution {
AMMExecution(UpdatedAmmModel),
pureExecution(InteractionData)
}

#[serde_as]
#[derive(Clone, Debug, Deserialize)]
pub struct SettledBatchAuctionModel {
    pub orders: HashMap<usize, ExecutedOrderModel>,
    #[serde(default)]
    pub executions: HashMap<usize, Execution>,
    pub ref_token: Option<H160>,
    #[serde_as(as = "HashMap<_, DecimalU256>")]
    pub prices: HashMap<H160, U256>,
}

```
But since this would introduce breaking chances and mix two completely different types, I decided against it.